File: coffeeshop/src/main/java/com/coffeeshop/coffeeshop/service/ReviewService.java
package com.coffeeshop.coffeeshop.service;

import com.coffeeshop.coffeeshop.model.Review;
import com.coffeeshop.coffeeshop.model.dto.request.ReviewUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface ReviewService {

    List<Review> findAll();

    Review getById(UUID id);

    Review create(Review entity);

    Review update(UUID id, Review entity);

    Review update(UUID id, Review entity, ReviewUpdateRequest request);

    void deleteById(UUID id);
}

File: coffeeshop-frontend/src/app/services/review.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { ReviewResponseDto, ReviewCreateRequest, ReviewUpdateRequest } from '../models/review.model';

@Injectable({ providedIn: 'root' })
export class ReviewService {
  private readonly http = inject(HttpClient);
  private readonly base = `${environment.apiUrl}/api/v1/review`;

  getAll(): Observable<ReviewResponseDto[]> {
    return this.http.get<ReviewResponseDto[]>(this.base);
  }

  getById(id: string): Observable<ReviewResponseDto> {
    return this.http.get<ReviewResponseDto>(`${this.base}/${id}`);
  }

  create(req: ReviewCreateRequest): Observable<ReviewResponseDto> {
    return this.http.post<ReviewResponseDto>(this.base, req);
  }

  update(id: string, req: ReviewUpdateRequest): Observable<ReviewResponseDto> {
    return this.http.put<ReviewResponseDto>(`${this.base}/${id}`, req);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}
